plugins/repository-azure/src/test/java/org/opensearch/repositories/azure/AzureRepositorySettingsTests.java

```diff
@@ -57,6 +58,7 @@
 public class AzureRepositorySettingsTests extends OpenSearchTestCase {
     @AfterClass
     public static void shutdownSchedulers() {
+        HttpResources.disposeLoopsAndConnections();
```

Comment:
@cwperks I am not sure we have to block here

Reply:
Thank you for fixing this @reta! I should have switched the PR to draft as I was unsure of this cleanup block. I saw that it fixed the thread leak, but didn't quite understand why.

I saw that it was stuck on NIOEventLoopGroup. Does this call remove all EventLoops in the group?

Reply:
> I saw that it was stuck on NIOEventLoopGroup. Does this call remove all EventLoops in the group?

It should, let's see if tests pass